<launch>
    <arg name="camera_namespace" default="camera_side"/>

    <node name="laplacian_line_detector" pkg="rr_iarrc" type="laplacian_line_detector"
          output="screen" ns="$(arg camera_namespace)">
        <!--Minimum Area to Keep-->
        <param name="min_blob_area" type="int" value="50" />
        
        <!--Range of the Laplacian Area to Floodfill-->
        <param name="laplacian_threshold_min" type="int" value="-255" />
        <param name="laplacian_threshold_max" type="int" value="-35" />

        <!--Strength of Floodfill-->
        <param name="Laplacian_threshold" type="int" value="30" />
        <!--Strength of Adaptive Thresholding-->
        <param name="ignore_adaptive" type="bool" value="True" /> <!-- improves results on side cameras -->
        <param name="adaptive_mean_threshold" type="int" value="0" />

        <!--The larger the height the lower it will be on the screen (Image Height = 1280)-->
        <param name="blockSky_height" type="int" value="0" />
        <param name="blockWheels_height" type="int" value="1280" />
        <param name="blockBumper_height" type="int" value="1280" />

        <param name="subscription_node" type="string" value="image_color_rect"/>
    </node>
</launch>
